### PR TITLE
Revise NEST 2.20.x guide on random numbers to be ajour with NEST 2.12 and later

### LINKED ISSUE #1816: NEST 2.4 random number information available in docs
**Describe the flaw**
It seems that the documentation [here](https://nest-simulator.readthedocs.io/en/stable/guides/random_numbers.html) is quite old. Is everything there still up to date? If so the NEST 2.4 version number that appears a few times is quite confusing. There's also an unformatted code block [here](https://nest-simulator.readthedocs.io/en/stable/guides/random_numbers.html#basics-of-parallel-simulation-in-nest). Are the examples still how one should go about RNGs in NEST 2.20?

### PR DESCRIPTION
**Pull request by heplesser**
This fixes #1816.

### CODE REVIEW

File: doc/guides/random_numbers.rst [hunk 1]
```diff
@@ -521,3 +247,198 @@ These commands should not generate any output. Obviously, this test
...
+
+.. _sli_generators:
+
+Generators available at the SLI level
+-------------------------------------
+
+This section is of historical interest only. Generators described here
+will no longer be available in NEST 3.0
+
+Random Numbers vs Random Deviates
```
Comment (jessica-mitchell) [inline doc/guides/random_numbers.rst:384]:
```suggestion
Random numbers vs random deviates
```

File: doc/guides/random_numbers.rst [hunk 2]
```diff
@@ -521,3 +247,198 @@ These commands should not generate any output. Obviously, this test
...
+| One RNG per VP in Python
+| These generators are used to randomized node properties (e.g., the
+  initial membrane potential) and connection properties (e.g., weights).
+
+The generators on the Python level are not strictly necessary, as one
+could in principle access the per-VP RNGs built into NEST. This would
+require very tedious SLI-coding, though. We therefore recommend at
+present that you use additional RNGs on the Python side.
+
+Why a Global RNG in NEST?
```
Comment (jessica-mitchell) [inline doc/guides/random_numbers.rst:350]:
```suggestion
Why a global RNG in NEST?
```

File: doc/guides/random_numbers.rst [hunk 3]
```diff
@@ -29,230 +29,34 @@ control RNGs in NEST.
...
-NEST thus creates divergent connections using a global RNG. This random
-number generator provides the exact same sequence of random numbers on
-each virtual process. Using this global RNG, each VP chooses 100 targets
-from the entire network, but actually creates connections only for those
-targets that reside on the VP. In practice, the global RNG is
-implemented using one "clone" on each VP; NEST checks occasionally that
-all these clones are synchronized, i.e., indeed generate identical
-sequences.
 
 Seeding the Random Generators
```
Comment (jessica-mitchell) [inline doc/guides/random_numbers.rst:61]:
```suggestion
Seeding the random generators
```

File: doc/guides/random_numbers.rst [hunk 4]
```diff
@@ -29,230 +29,34 @@ control RNGs in NEST.
...
+   
+   NEST 3.0 will include a significantly revised and more convenient
+   interface to random numbers.
 
--  ``uniformint``
--  ``normal_clipped``, ``normal_clipped_left``, ``normal_clipped_right``
+.. _seeding:
 
-generators from NEST 2.2 need to be adapted as detailed below.
+Seeding and Selecting Random Generators
```
Comment (jessica-mitchell) [inline doc/guides/random_numbers.rst:48]:
```suggestion
Seeding and selecting random generators
```

File: doc/guides/random_numbers.rst [hunk 5]
```diff
@@ -521,3 +247,198 @@ These commands should not generate any output. Obviously, this test
...
+generators:
+
+| The global NEST RNG
+| This generator is mainly used when creating connections using
+  ``fixed_outdegree`` connection rule.
+
+| One RNG per VP in NEST
+| These generators are used when creating connections using
+  most other probabilistic connection rules and to provide random numbers to nodes
+  generating random output, e.g. the ``poisson_generator``.
```
Comment (jessica-mitchell) [inline doc/guides/random_numbers.rst:339]:
```suggestion
  generating random output, e.g., the ``poisson_generator``.
```

**Review by jessica-mitchell — CHANGES_REQUESTED** [review]
Thanks for updating these docs. I have a question regarding the sli code in sections "Setting a different global RNG" and "Setting different per-processes RNGs" - is there not be a way to do this with Python?

**Comment by heplesser** [comment]
> Thanks for updating these docs. I have a question regarding the sli code in sections "Setting a different global RNG" and "Setting different per-processes RNGs" - is there not be a way to do this with Python?

No, it isn't, one of the long-lasting weaknesses of NEST 2. But all is solved in NEST 3 (#1549).

File: doc/guides/connection_management.rst [hunk 6]
```diff
@@ -3,19 +3,6 @@
...
-However, simple cases, which are just creating one-to-one connections
-between two lists of nodes are still working with the new command
-without the need to change the code. Note that the topology-module is
-not effected by theses changes. The translation between the old and the
-new connect routines is described in `Old Connection
-Routines <connection-management.md#old-connection-routines>`__.
-
 The connectivity pattern is defined inside the `Connect()` function
 under the key 'rule'. The patterns available are described in
 `Connection Rules <connection-management.md#connection-rules>`__. In
```
[the remark this answers is not in the document]
Comment (heplesser) [reply doc/guides/connection_management.rst]: I have fixed the link.

**Comment by heplesser** [comment]
MacOS tests will fail because NumPy cannot be installed with the 2.20 Travis setup. This problem is *wontfix*, so failing macOS tests **do not** block merging of this PR.

**Review by jessica-mitchell — APPROVED** [review]
Looks good!

**Pull request merged**